bokehjs/src/lib/models/widgets/radio_group.ts

```diff
@@ -21,7 +22,10 @@ export class RadioGroupView extends InputGroupView {
     this._inputs = []
     for (let i = 0; i < labels.length; i++) {
       const radio = input({type: "radio", name, value: `${i}`})
-      radio.addEventListener("change", () => this.change_active(i))
+      radio.addEventListener("click", () => {
+        this.change_active(i)
+        this.model.trigger_event(new ButtonClick())
```

Comment:
Radios and checkboxes aren't buttons, just stateful widgets. Checking their `active` property change should be sufficient. Besides this, `click` is much more limited event that `change`, degrading functionality of this widget.

Reply:
They had `on_click` already for some reason, though. If you are OK just dropping those entirely at 3.0 (just Radios and Checkboxes) then this could be dispensed with.

Reply:
The original motivation behind this PR was to allow Toggle buttons to have their state set programatically without invoking the click callback - which if you had an `on_click` set on a Toggle, setting it's `active` state would cause the click callback to fire.

If we drop click events on this radio/checkbox group buttons, the above issue would remain with these widgets. I can buy the argument that radio/checkboxes are strictly stateful widgets, and a click event is not useful, however, I'd only point out that you can infact have click and change handlers on them in Javascript land.

I personally don't use radios/checkboxes very often so I'm not strongly opinionated either way.

Reply:
I guess the question is whether there is any demand to be able to differentiate real UI clicks of "stateful buttons" from programmatic assignment in the same way as "plain buttons". I don't have strong feelings of preferences about that. These options for "stateful buttons" would be ok to me:

* have them emit `ButtonClick` and have them keep a (deprecated) `on_click` that responds to the _event_ (to be consistent with every other `on_click` at this point)
* have them not emit `ButtonClick`, but remove `on_click` immediately

What I would be 👎  on would be to have some object emit `ButtonClick` but not have an `on_click` (or vice versa) or to have some `on_click` methods mean different things than other `on_click` methods.

Reply:
I'm inclined to just remove the `on_click` and revert the Javascript change here (option 2).

Reply:
That seems to be what @mattpap also desires, so 👍 from me to  proceed with that (just need a minimal note in the release notes about the changes: https://github.com/bokeh/bokeh/blob/branch-3.0/sphinx/source/docs/releases/3.0.0.rst) Thanks for your patience with some back and forth @xaptronic !

Reply:
>I guess I am + 0.5 to keep these other "stateful buttons" doing the same, for that reason.

Will proceed to remove the `on_click` from the radio/checkbox group buttons, but let's not forget about your comment here, as it could be something to revisit in the future. Perhaps there is some wisdom to having the events be similar to what you can do natively in the browser, or having some kind of system where you can freely attach any browser event to an event and listen on the python side - much larger discussion way outside the scope of these changes, and certainly more than enough on the queue for now :)